README.md
# assemble-template

> Some assembly required.

## [Assemble](http://assemble.io/)

Assemble is a component and static site generator that makes it dead simple to build modular sites, documentation and components from reusable templates and data.

* Documentation
* Plugins - Plugins extend the core functionality of Assemble.
* Helpers - Documentation for the helpers in the [handlebars-helpers](http://github.com/assemble/handlebars-helpers) library.

## Contributing
All contributions are welcome! The simplest way to show your support for this project is to **"star" it**. Please see [Contributing to Assemble](http://assemble.io/contributing) for more information.

## Release History
 * 2014-07-16   v0.1.0   Generated by the [Yeoman Generator](https://github.com/assemble/generator-assemble) for [Assemble](http://assemble.io)

## Special Thanks to:

- Icomoon
- Waypoints.js
- Animate.css
- Fontawesome


CHANGELOG
v0.1.0:
  date: "2014-07-16"
  changes:
    - Generated by the [Yeoman Generator](https://github.com/assemble/generator-assemble) for [Assemble](http://assemble.io)


compass.rb
# Require any additional compass plugins here.

# Set this to the root of your project when deployed:
http_path = "/"
css_dir = "dist/assets/css"
sass_dir = "src/assets/scss"
images_dir = "src/assets/img"
javascripts_dir = "dist/assets/js"

# You can select your preferred output style here (can be overridden via the command line):
# output_style = :expanded or :nested or :compact or :compressed

output_style = :expanded


# To enable relative paths to assets via compass helper functions. Uncomment:
# relative_assets = true

# To disable debugging comments that display the original location of your selectors. Uncomment:
# line_comments = false

line_comments = true

asset_cache_buster :none


# If you prefer the indented syntax, you might want to regenerate this
# project again passing --syntax sass, or you can uncomment this:
# preferred_syntax = :sass
# and then run:
# sass-convert -R --from scss --to sass sass scss && rm -rf sass && mv scss sass

module Sass::Script::Functions
  def random(max = Sass::Script::Number.new(100))
    Sass::Script::Number.new(rand(max.value), max.numerator_units, max.denominator_units)
  end
end

src/assets/js/scripts.js
/***************** Slide-In Nav ******************/

$(window).load(function() {

    $('.nav_slide_button').click(function() {
        $('.flip-nav').toggleClass('open');
    });

});

document.querySelector("#nav-toggle").addEventListener("click", function() {
    this.classList.toggle("active");
});



$(document).ready(function(){
    /*  
    Set up waypoints (Waypoints.js)
    https://github.com/imakewebthings/jquery-waypoints
    */
    $('.waypoint-1').waypoint(function() {
        $('.waypoint-1 #title').addClass('animated fadeInDown');
        $('.waypoint-1 #sub-title').addClass('animated fadeInUp delay-short');
    }, {
        offset: '75%'
    });

    $('.waypoint-2').waypoint(function() {
        $('#section-a .anchor').addClass('animated zoomIn');
        $('#section-a h3').addClass('animated fadeInUp');
        $('#section-a p').addClass('animated fadeInUp delay-short');
    }, {
        offset: '75%'
    });
    $('.waypoint-3').waypoint(function() {
        $('.waypoint-3.photo-wrapper').addClass('animated zoomIn delay-short');
    }, {
        offset: '75%'
    });

    $('.waypoint-4').waypoint(function() {
        
        $('.waypoint-4 h4').addClass('animated fadeIn');
        $('.waypoint-4 p').addClass('animated fadeIn delay-short');
        $('.waypoint-4 .large').addClass('animated fadeInDown');
    }, {
        offset: '75%'
    });

    /* instantiate Stellar.js (parallax) */
    $(window).stellar();
});








Gruntfile.js
  /*
 * Generated on 2014-07-16
 * generator-assemble v0.4.10
 * https://github.com/assemble/generator-assemble
 *
 * Copyright (c) 2014 Hariadi Hinta
 * Licensed under the MIT license.
 */

'use strict';

// # Globbing
// for performance reasons we're only matching one level down:
// '<%= config.src %>/templates/pages/{,*/}*.hbs'
// use this if you want to match all subfolders:
// '<%= config.src %>/templates/pages/**/*.hbs'

module.exports = function(grunt) {

  require('time-grunt')(grunt);

  // Project configuration.
  grunt.initConfig({

    config: {
      src: 'src',
      dist: 'dist'
    },
    compass: {
      dist: {
        options: {
          config: 'compass.rb',
          force: true
        }
      }
    },
    autoprefixer: {
        single_file: {
            src: '<%= config.dist %>/assets/css/styles.css',
            dest: '<%= config.dist %>/assets/css/styles.css'
        }
    },
    copy: {
      javascript: {
        files: [
          {
            expand: true,
            flatten: false,
            cwd: '<%= config.src %>/assets/js/',
            src: ['**/*.js'],
            dest: '<%= config.dist %>/assets/js/'
          }
        ]
      }
    },
    watch: {
      assemble: {
        files: ['<%= config.src %>/{content,data,templates}/{,*/}*.{md,hbs,yml}'],
        tasks: ['assemble']
      },
      copy: {
        files: ['<%= config.src %>/assets/js/{,*/}*.js'],
        tasks: ['copy']
      },
      autoprefixer: {
        files: ['<%= config.src %>/assets/scss/**/{,*/}*.scss'],
        tasks: ['autoprefixer']
      },
      compass: {
         files: ['<%= config.src %>/assets/scss/**/{,*/}*.scss'],
         tasks: ['compass', 'autoprefixer'],
         options: { livereload: true }
      },
      livereload: {
        options: {
          livereload: '<%= connect.options.livereload %>'
        },
        files: [
          '<%= config.dist %>/{,*/}*.html',
          '<%= config.dist %>/assets/{,*/}*.css',
          '<%= config.dist %>/assets/{,*/}*.js',
          '<%= config.dist %>/assets/{,*/}*.{png,jpg,jpeg,gif,webp,svg}'
        ]
      }
    },

    connect: {
      options: {
        port: 9000,
        livereload: 35729,
        // change this to '0.0.0.0' to access the server from outside
        hostname: 'localhost'
      },
      livereload: {
        options: {
          open: true,
          base: [
            '<%= config.dist %>'
          ]
        }
      }
    },

    assemble: {
      pages: {
        options: {
          flatten: true,
          assets: '<%= config.dist %>/assets',
          layout: '<%= config.src %>/templates/layouts/default.hbs',
          data: '<%= config.src %>/data/*.{json,yml}',
          partials: '<%= config.src %>/templates/partials/*.hbs'
        },
        files: {
          '<%= config.dist %>/': ['<%= config.src %>/templates/pages/*.hbs']
        }
      }
    },

    // Before generating any new files,
    // remove any previously-created files.
    clean: ['<%= config.dist %>/**/*.{html,xml}']

  });

  grunt.loadNpmTasks('assemble');
  grunt.loadNpmTasks( 'grunt-contrib-compass' );
  grunt.loadNpmTasks('grunt-contrib-clean');
  grunt.loadNpmTasks('grunt-contrib-copy');
  grunt.loadNpmTasks('grunt-contrib-connect');
  grunt.loadNpmTasks('grunt-contrib-watch');
  grunt.loadNpmTasks('grunt-autoprefixer');


  grunt.registerTask('server', [
    'clean',
    'assemble',
    'connect:livereload',
    'watch',
    'copy',
  ]);

  grunt.registerTask('build', [
    'clean',
    'assemble',
    'autoprefixer',
  ]);

  grunt.registerTask('default', [
    'build'
  ]);

};
